decoder/src/log/stdout_logger.rs

```diff
@@ -0,0 +1,310 @@
+use colored::{Color, Colorize};
+use dissimilar::Chunk;
+use log::{Level, Log, Metadata, Record as LogRecord};
+
+use std::{
+    fmt::Write as _,
+    io::{self, StderrLock, StdoutLock},
+    path::Path,
+    sync::atomic::{AtomicUsize, Ordering},
+};
+
+use super::format;
+use super::{format::LogSegment, DefmtRecord};
+
+enum Record<'a> {
+    Defmt(&'a DefmtRecord<'a>),
+    Host(&'a LogRecord<'a>),
+}
+
+pub(crate) struct StdoutLogger {
```

Comment:
I agree that `StdoutLogger` is a better term. Since the renaming of the file and struct messes up the diff and makes reviewing a bit harder, I will change the name in a separate PR and ask you to please rebase your PR on that.

Reply:
Done